server/handlers/get_totp.go

```diff
@@ -0,0 +1,46 @@
+package handlers
+
+import (
+	"bytes"
+	"encoding/base64"
+	"image/png"
+	"net/http"
+
+	"github.com/keratin/authn-server/app"
+	"github.com/keratin/authn-server/app/services"
+	"github.com/keratin/authn-server/lib/route"
+	"github.com/keratin/authn-server/server/sessions"
+)
+
+//GetTOTP begins the TOTP onboarding process
+func GetTOTP(app *app.App) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// check for valid session with live token
+		accountID := sessions.GetAccountID(r)
+		if accountID == 0 {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		totpKey, err := services.TOTPCreator(app.AccountStore, app.TOTPCache, accountID, route.MatchedDomain(r))
+		if err != nil {
+			panic(err)
+		}
+
+		//Convert to png
+		var pngImg bytes.Buffer
+		img, err := totpKey.Image(200, 200)
+		if err != nil {
+			panic(err)
+		}
+		err = png.Encode(&pngImg, img)
```

Comment:
I think we need to at least send the secret in text form here so that users can manually onboard (eg if I am already on the device) or save the code.

I feel like there is a good case to be made for removing the image entirely and leaving generation to the caller so they can customize size and other details as needed.

Reply:
An example flow here would be to turn the secret itself into a URI along these lines that includes a lot of app-specific (from authn context) detail to identify the creds in the user's app.

https://github.com/google/google-authenticator/wiki/Key-Uri-Format

The URI can then be 

1. clickable if the user is on the phone
2. URI from (1) can be turned into a QR code using eg react-QR-code

Additionally you could allow the user to copy the secret for backup purposes or manual onboarding.

I think my preference would be that authn doesn't know how to do any of this.

Reply:
The URL we get back here is pretty reasonable and comes free with the package so probably no harm in returning that.  If we do want to provide encoded images config overrides for 

1. whether to return the image at all 
2. image dimensions
3. image format (eek) 

Might be helpful.  Open to ways we could surface through the API as well (eg only return an image if dimensions are provided).